### ISSUE TITLE
Vulkan backend cannot run on Mac unless manually installing MoltenVK or Vulkan SDK

### ISSUE THREAD

**Issue by lin-hitonami** [taichi-dev/taichi#3787]
**Describe the bug**
Although taichi now ships a copy of `libMoltenVK.dylib` in the package, it is not correctly used by `taichi_core.so`. By executing `otool -L` on `taichi_core.so`, we can see that `taichi_core.so` uses the dylib in the homebrew directory instead the one in the package. Following is the output:
```
taichi_core.so:
        @rpath/libtaichi_core.dylib (compatibility version 0.0.0, current version 0.0.0)
        /System/Library/Frameworks/Cocoa.framework/Versions/A/Cocoa (compatibility version 1.0.0, current version 23.0.0)
        /System/Library/Frameworks/IOKit.framework/Versions/A/IOKit (compatibility version 1.0.0, current version 275.0.0)
        /System/Library/Frameworks/CoreFoundation.framework/Versions/A/CoreFoundation (compatibility version 150.0.0, current version 1775.118.101)
        /usr/lib/libz.1.dylib (compatibility version 1.0.0, current version 1.2.11)
        /usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1292.100.5)
        /opt/homebrew/opt/molten-vk/lib/libMoltenVK.dylib (compatibility version 1.0.0, current version 1.0.0)
        /System/Library/Frameworks/Metal.framework/Versions/A/Metal (compatibility version 1.0.0, current version 244.119.0)
        /opt/homebrew/opt/llvm@11/lib/libc++.1.dylib (compatibility version 1.0.0, current version 1.0.0)
        /System/Library/Frameworks/AppKit.framework/Versions/C/AppKit (compatibility version 45.0.0, current version 2022.44.149)
        /System/Library/Frameworks/CoreGraphics.framework/Versions/A/CoreGraphics (compatibility version 64.0.0, current version 1463.4.2)
        /System/Library/Frameworks/CoreServices.framework/Versions/A/CoreServices (compatibility version 1.0.0, current version 1122.33.0)
        /System/Library/Frameworks/Foundation.framework/Versions/C/Foundation (compatibility version 300.0.0, current version 1775.118.101)
        /usr/lib/libobjc.A.dylib (compatibility version 1.0.0, current version 228.0.0)
```
Executing `install_name_tool -change /opt/homebrew/opt/molten-vk/lib/libMoltenVK.dylib @rpath/libMoltenVK.dylib taichi_core.so` after `taichi_core.so` is produced helps solving this problem.

**Comment by ailzhang** [comment]
cc: @bobcao3 
Maybe worth trying https://cmake.org/pipermail/cmake/2006-June/009758.html? 

In the worst case this can be fixed by adding a post build step, similar to https://coderedirect.com/questions/546925/how-do-you-add-an-rpath-to-an-executable-in-cmake-on-build-not-install-on-osx.

**Comment by lin-hitonami** [comment]
fixed in #4205
